File: java_samples/payloads/request_signature.java
package payloads;
import java.util.ArrayList;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.entity.StringEntity;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClients;
import org.json.JSONObject;

public class request_signature {
	
	private int template_file_id;
	private boolean is_ordered;
	private String name;
	private ArrayList<recipients> recipients = new ArrayList<recipients>();
	private String message = "";
	private ArrayList<cc> cc = new ArrayList<cc>();
	private String file_password = "";
	private boolean embedded_signing = false;
	private ArrayList<merge_fields> merge_fields = new ArrayList<merge_fields>();

	public request_signature(int template_file_id, String name, boolean is_ordered, recipients recip) 
	{
		this.template_file_id = template_file_id;
		this.is_ordered = is_ordered;
		this.name  = name;
		this.recipients.add(recip);
	}
	
	public void set_message(String msg) 
	{
		this.message = msg;
	}
	
	public void add_cc(cc obj) 
	{
		this.cc.add(obj);
	}
	
	public void set_file_password(String msg) 
	{
		this.file_password = msg;
	}
	
	public void set_embedded_signing(boolean msg) 
	{
		this.embedded_signing = msg;
	}
	
	public void add_merge_field(merge_fields obj) 
	{
		this.merge_fields.add(obj);
	}
	
	public void add_recipients(recipients obj) 
	{
		this.recipients.add(obj);
	}
	
	public JSONObject toJson() 
	{
		JSONObject obj = new JSONObject();

		ArrayList<JSONObject> array = new ArrayList<JSONObject>();
		
		for (int i = 0; i < this.recipients.size(); i++)  
		{
            array.add(this.recipients.get(i).toJson());         
		} 
		obj.put("recipients", array);
		array.clear();
		
		for (int i = 0; i < this.cc.size(); i++)  
		{
            array.add(this.cc.get(i).toJson());         
		} 
		obj.put("cc", array);
		array.clear();
		
		for (int i = 0; i < this.merge_fields.size(); i++)  
		{
            array.add(this.merge_fields.get(i).toJson());         
		} 
		obj.put("merge_fields", array);
		array.clear();
		
		obj.put("template_file_id", this.template_file_id);
		obj.put("is_ordered", this.is_ordered);
		obj.put("name", this.name);
		obj.put("message", this.message);
		
		
		obj.put("file_password", this.file_password);
		obj.put("embedded_signing", this.embedded_signing);
		
		return obj;
		
	}
	
	public CloseableHttpResponse create_signature_templates(String base_url, String api_token) throws Exception {

        HttpPost post = new HttpPost(base_url + "v1/files/pending/template/");
        post.addHeader("content-type", "application/json");
        post.addHeader("Authorization", api_token);
        post.setEntity(new StringEntity(this.toJson().toString()));

        CloseableHttpClient httpClient = HttpClients.createDefault();
        CloseableHttpResponse response = httpClient.execute(post);

        return response;
    }


}

File: v2/node_sample/payload/RequestSignature.js
const Recipients = require('./Recipients');
const CC = require('./CC');
const MergeFields = require('./MergeFields');
var unirest = require('unirest');

class RequestSignature
{
    message = ''
    cc = []
    file_password = ''
    embedded_signing = false
    merge_fields = []
    template_file_id = 0
    recipients = []
    is_ordered = false
    name = ''

    constructor(template_file_id, recipients, is_ordered, name, message='', cc=null, file_password='',
                embedded_signing=false, merge_fields=null)
  {
    this.template_file_id = template_file_id;
    this.add_recipients(recipients);
    this.is_ordered = is_ordered;
    this.name = name;
    this.message = message;

    if (cc != null)
    {
        this.add_cc(cc)
    }
    this.file_password = file_password;
    this.embedded_signing = embedded_signing;

    if (merge_fields != null)
    {
        this.add_merge_fields(merge_fields);
    }
  }

  add_recipients(recipients)
  {
    if (!(recipients instanceof Recipients))
    {
        throw new Error("Invalid type of recipients");
    }
    this.recipients.push(recipients);
  }

  add_cc(cc)
  {
    if (!(cc instanceof CC))
    {
        throw new Error("Invalid type of cc");
    }
    this.cc.push(cc);
  }

  add_merge_fields(merge_fields)
  {
    if (!(merge_fields instanceof MergeFields))
    {
        throw new Error("Invalid type of merge_fields");
    }
    this.merge_fields.push(merge_fields);
  }

  to_json()
  {
        return  {"template_file_id": this.template_file_id, "is_ordered": this.is_ordered, "name": this.name,
                 "recipients": this.recipients, "cc": this.cc, "message": this.message, "merge_fields": this.merge_fields,
                 "file_password": this.file_password, "embedded_signing": this.embedded_signing}
  }

  create_signature_templates = function (base_url, api_token) {
        var payload = this.to_json();
        return new Promise((resolve, reject) => {
        unirest.post(base_url + `v2/template/${this.template_file_id}/rs`)
        .headers({'Authorization': api_token, 'content-type': "application/json"})
        .send(payload)
        .end(function (response) {
        if (response.error) {
        return reject(response.body)
        }
        return resolve(response.body);
        });
    })
    }
}

module.exports = RequestSignature
/*
try{
obj = new RequestSignature(0, "a", false, "c");
console.log(obj.to_json());
}
catch (e){
console.log(e.message);
}*/